src/include/basil.h

```diff
@@ -353,15 +405,18 @@ typedef enum {
 	basil_1_1,
 	basil_1_2,
 	basil_1_3,
-	basil_1_4
+	basil_1_4,
+	basil_1_5, /* not supported */
+	basil_1_6, /* not supported */
+	basil_1_7
```

Comment:
If I remove basil_1_5 and basil_1_6 , we are still left with basil_1_7. This will make switch cases to have basil_1_7 case where it is not valid, even though the case greater then basil_1_4 will never hit with existing code. I don't see much value in removing this part of the code.

Reply:
Well one case I can think of is that if response from Basil comes out to be with faulty version (1.5 or 1.6) then because of these switch statements we might end up parsing it and possibly fail/crash.
So I think it is ok to remove them

Reply:
@dilip-krishnan  this is the comment I was talking about, what do you think?

Reply:
> Well one case I can think of is that if response from Basil comes out to be with faulty version (1.5 or 1.6)
Actually  that won't be possible, because we do querry engine only through supported version. We loop through pbs_supported_basil_version to check, when we get a status success in return from query using that particular version. So i can assure that no matter how messed the reply of basil comes out, the version will be always be either set supported version or will end up in failure. This code exists in alps_engine_query and alps_engine_query_knl.

I don't see any harm in both keeping and removing this part of code. If you still feel that I should remove it, then I'll do it.